### DISCUSSION TITLE
Was there a change in how to Run a Shortcut using a Shell Script?

### DISCUSSION THREAD

**Discussion by SteveMattan** [ttscoff/bunch-docs#297] [Q&A] [answered]
I just recently upgraded from Monterey to Ventura (13.3.1). I had been using a the following to turn on/off speakers plugged into a smart outlet.

# This works in macOS 12.x but not in macOS 13.3.1
$ /usr/bin/shortcuts run "Turn On Speakers"
!$ /usr/bin/shortcuts run "Turn Off Speakers"

When I run this script via the "Shell Scripts" menu in Bunch I get the following error to STOUT:

/bin/sh: /usr/bin.shortcuts run: No such file or directory

When I run the  command "/usr/bin/shortcuts run "Turn On Speakers"" in the terminal, it works just fine.

As mentioned above I'm running macOS 13.3.1. And I'm running Bunch 1.4.12 (172).

Did something change such that the command in Bunch needs to be updated?

Note that I have changed my Bunch file to use:

(sc Turn On Speakers)
!(sc Turn Off Speakers)

which has the desired outcome. 

Thanks,

SteveM

**Comment by ttscoff** [comment]
Huh, that's really weird that it's replacing the slash with a dot in 
your shell command, I'll have to look into it. But for now the ideal 
solution is what you discovered, using the `(shortcut NAME)` command.

-Brett

**Comment by SteveMattan** [comment]
Yeah, it's not replacing the slash with a dot. That's just my inability to type! 

Sorry about that. Here is a screenshot showing the Bunch file (in BBEdit), iTerm showing that things seem to be normal, and the error window. Hope this helps. (And that's the wrong screenshot (intermediate notes to myself ...)! But it is the correct error window so I'm leaving it.)

And again sorry about that. 

And thank you for the quick response.

![Screenshot 2023-04-16 at 4 32 56 PM](https://user-images.githubusercontent.com/95952224/232343657-79a0dca2-a23a-4162-9ac4-9944057e4d1d.png)

**Comment by ttscoff** [comment]
Ok, I'm able to replicate this but not able to explain it, will have to 
do some digging. It's somehow interpreting "run" as the command and not 
an argument. In other commands I run that have multiple arguments, they 
work fine, so this is a weird one.

One thing that _does_ work is to write a simple shell script like:

```
#!/bin/bash

shortcuts run "$@"
```

And then in the Bunch:

```
$ ~/scripts/sc_test.sh "Name of Shortcut"
```

But I'll look into why this particular command is doing this, as it 
probably affects other commands as well.

-Brett

**Comment by ttscoff** [comment]
Something is capturing the `/usr/bin/` and interpreting it as a 
processor. This works fine:

$ shortcuts run "name of shortcut"

I'm just making a record of the debugging, I know you already have the 
(sc) command solution.

-Brett

**Comment by ttscoff** [comment, marked as answer]
Found it and fixed for the next update. The method that was trying to 
figure out a shebang for a script wasn't recognizing the executable as a 
binary and when it got an error trying to read it as a text file, it was 
returning nil, which then got replaced with `/bin/sh -c` but lost the 
executable in the process. That was just a boner on my part.

-Brett

**Comment by SteveMattan** [comment]
Great!

And sorry again about that stupid typo. I’m glad you responded or else you would have been chasing an issue that wasn’t.

Stay well,

SteveM

**Comment by ttscoff** [comment]
No problem, and thanks for the donation :)
